core/src/main/java/org/infinispan/container/entries/ReadCommittedEntry.java

```diff
@@ -195,9 +213,10 @@ public long getLastUsed() {
    }
 
    @Override
-   public Object setValue(Object value) {
-      Object prev = this.value;
-      this.value = value;
+   public Object setValue(Object newValue) {
+      Object prev = oldValue;
+      this.oldValue = this.value;
+      this.value = newValue;
```

Comment:
Not sure how this is supposed to work when a tx writes to the same key 2 times, 3 times, etc. Personally I'd prefer an explicit update of `oldValue` and `setValue` always returning the current value (aka `this.value`) -- something `CallInterceptor.performPut()` already does?

At least the entry is removed from the context when a command is retried, so you don't have to worry about non-tx caches.

Reply:
hmmm... I wasn't expecting the "old value" to be updated every time.

Reply:
reverted